src/metaxy/metadata_store/duckdb.py

```diff
@@ -165,3 +187,32 @@ def _create_native_components(self):
         diff_resolver = NarwhalsDiffResolver()
 
         return joiner, calculator, diff_resolver
+
+    # ------------------------------------------------------------------ DuckLake
+    def open(self) -> None:
+        """Open DuckDB connection and configure optional DuckLake attachment."""
+        super().open()
+        if self._ducklake_attachment is not None:
+            if self._conn is None:
+                raise RuntimeError(
+                    "DuckDBMetadataStore failed to open DuckDB connection."
+                )
+            self._ducklake_attachment.configure(self._conn)
+
+    def preview_ducklake_sql(self) -> list[str]:
+        """Return DuckLake attachment SQL if configured."""
+        if self._ducklake_attachment is None:
```

Comment:
It's better to make a public property that does the check internally and removes the `Optional` case: 

```py
@property
def ducklake_attachment(self) -> DucklakeAttachment:
    if not self._ducklake_attachment is None:
        raise ..
    else:
        reuturn self._ducklake_attachment
```

This will simplify using this attribute in places where we **know** it should exist. 

Same should be done with other optional fields.